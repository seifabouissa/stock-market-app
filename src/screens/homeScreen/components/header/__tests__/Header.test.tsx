import React from 'react';
import ReactTestRenderer from 'react-test-renderer';
import Header from '..';
import {Image, TextInput} from 'react-native';

const imagePath = '../../../src/assets/images/nasdaq.png';

jest.mock('react-native-config', () => ({
  API_URL: 'https://api.polygon.io/v3/reference/tickers',
  API_KEY: 'WVKkJbUc3M82eMzGr_dV77F4lRfrkBXi',
}));

jest.mock('@react-native-async-storage/async-storage', () => ({
  getItem: jest.fn(),
  setItem: jest.fn(),
}));

describe('Header component renders correctly', () => {
  it('Header component renders correctly', () => {
    const renderer = ReactTestRenderer.create(<Header />);
    const instance = renderer.root;
    const imageElement = instance.findByType(Image);
    const textInputElement = instance.findByType(TextInput);
    const imageSource = imageElement.props.source.testUri;
    const inputPlaceholder = textInputElement.props.placeholder;
    expect(imageSource).toBe(imagePath);
    expect(inputPlaceholder).toBe('Search');
  });

  it('Check if header component matches the snapshot', () => {
    const {toJSON} = ReactTestRenderer.create(<Header />);
    const tree = toJSON();
    expect(tree).toMatchSnapshot();
  });
});
