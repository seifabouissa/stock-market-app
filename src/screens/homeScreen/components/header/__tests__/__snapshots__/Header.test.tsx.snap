// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders correctly Check if header component matches the snapshot 1`] = `
<View
  style={
    {
      "backgroundColor": "#576682",
      "borderBottomLeftRadius": 8,
      "borderBottomRightRadius": 8,
      "marginBottom": 20,
      "paddingBottom": 15,
      "paddingHorizontal": 20,
    }
  }
>
  <Image
    resizeMode="contain"
    source={
      {
        "testUri": "../../../src/assets/images/nasdaq.png",
      }
    }
    style={
      {
        "height": 50,
        "width": 100,
      }
    }
  />
  <TextInput
    onChangeText={[Function]}
    placeholder="Search"
    placeholderTextColor="#000"
    style={
      {
        "backgroundColor": "#fff",
        "borderRadius": 8,
        "padding": 10,
      }
    }
  />
</View>
`;
